File: javabackend/src/main/java/mft/dev/javabackend/dto/response/UtenteResponseDTO.java
package mft.dev.javabackend.dto.response;

import com.fasterxml.jackson.annotation.JsonProperty;
import mft.dev.javabackend.model.Utente;

import java.util.List;

public record UtenteResponseDTO(
        @JsonProperty(value = "id", required = true)
        long id,
        @JsonProperty(value = "nome", required = true)
        String nome,
        @JsonProperty(value = "cognome", required = true)
        String cognome,
        @JsonProperty("email")
        String email,
        @JsonProperty("todos")
        List<TodoResponseDTO> todos
) {
    public static UtenteResponseDTO fromUtente(Utente utente){
        return new UtenteResponseDTO(
                utente.getId(),
                utente.getNome(),
                utente.getCognome(),
                utente.getEmail(),
                utente.getTodoList().stream().map(TodoResponseDTO::fromTodo).toList()
        );
    }
}

File: kotlinbackend/src/main/kotlin/mft/dev/kotlinbackend/dto/response/UtenteResponseDTO.kt
package mft.dev.kotlinbackend.dto.response

import com.fasterxml.jackson.annotation.JsonProperty
import mft.dev.kotlinbackend.model.Utente

data class UtenteResponseDTO(
    @JsonProperty(value = "id", required = true)
    val id: Long,
    @JsonProperty(value = "nome", required = true)
    val nome: String,
    @JsonProperty(value = "cognome", required = true)
    val cognome: String,
    @JsonProperty("email")
    val email: String?,
    @JsonProperty("todos")
    val todos: MutableList<TodoResponseDTO>
) {
    companion object {
        fun fromUtente(utente: Utente): UtenteResponseDTO {
            return UtenteResponseDTO(
                utente.id!!,
                utente.nome,
                utente.cognome,
                utente.email,
                utente.todoList.map { TodoResponseDTO.fromTodo(it) }.toMutableList()
            )
        }
    }
}
